File: video/v1/composition_settings/create-upload-to-external-s3/create-upload-to-external-s3.5.x.cs
// Install the C# / .NET helper library from twilio.com/docs/csharp/install

using System;
using Twilio;
using Twilio.Rest.Video.V1;


class Program 
{
    static void Main(string[] args)
    {
        // Find your Account Sid and Token at twilio.com/console
        // DANGER! This is insecure. See http://twil.io/secure
        const string accountSid = "ACXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX";
        const string authToken = "your_auth_token";

        TwilioClient.Init(accountSid, authToken);

        var compositionSettings = CompositionSettingsResource.Create(
            awsS3Url: new Uri("https://my-bucket.s3.amazonaws.com/recordings"),
            awsStorageEnabled: true,
            awsCredentialsSid: "CRXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX",
            friendlyName: "Upload to external bucket"
        );

        Console.WriteLine(compositionSettings.FriendlyName);
    }
}

File: video/v1/composition_settings/create-upload-to-external-s3/create-upload-to-external-s3.5.x.php
<?php

// Update the path below to your autoload.php,
// see https://getcomposer.org/doc/01-basic-usage.md
require_once '/path/to/vendor/autoload.php';

use Twilio\Rest\Client;

// Find your Account Sid and Auth Token at twilio.com/console
// DANGER! This is insecure. See http://twil.io/secure
$sid    = "ACXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX";
$token  = "your_auth_token";
$twilio = new Client($sid, $token);

$composition_settings = $twilio->video->v1->compositionSettings()
                                          ->create("Upload to external bucket", // friendlyName
                                                   array(
                                                       "awsS3Url" => "https://my-bucket.s3.amazonaws.com/recordings",
                                                       "awsStorageEnabled" => True,
                                                       "awsCredentialsSid" => "CRXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX"
                                                   )
                                          );

print($composition_settings->friendlyName);